HW05.Rmd

```diff
@@ -0,0 +1,133 @@
+---
+title: "HW05"
+output: github_document
+author: Bei Wang
+---
+
+```{r load pkgs, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library("tidyverse")
+library("magrittr")
+library("knitr")
+library("readr")
+```
+
+## 1. Tidying the dad_mom file
+
+    Each variable must have its own column.
+    Each observation must have its own row.
+    Each value must have its own cell.
+
+```{r dad_mom}
+#read dataset
+dad_mom<- read.table("dad_mom.txt", header=TRUE) 
+
+#reshape
+tidy_dad_mom<- dad_mom %>%
+  unite("dad", name_dad, income_dad) %>%
+  unite("mom", name_mom, income_mom) %>%
+  pivot_longer(c(dad, mom), names_to="parent", values_to="income") %>%
+  separate(income, c("name", "income"))
+
+#print
+tidy_dad_mom
+```
+
+## 2. Joining together CTRP data
+
+```{r totalauc}
+#read files
+AUC_1 <- read_csv("CTRP_files/AUC_1.csv")
+AUC_2 <- read_csv("CTRP_files/AUC_2.csv")
+cancer_cell_line_info <- read_csv("CTRP_files/cancer_cell_line_info.csv")
+compound_info <- read_csv("CTRP_files/compound_info.csv")
+Experiment_info <- read_csv("CTRP_files/Experiment_info.csv")
+
+#glimpse
+glimpse(AUC_1)
+glimpse(AUC_2)
+glimpse(cancer_cell_line_info)
+glimpse(compound_info)
+glimpse(Experiment_info)
```

Comment:
Didn't know this useful function for previewing the data, I literally clicked on every dataset for their detailed information lol

Reply:
yep! I found this and the skim() from the skimr package are very useful for understanding a dataset.